code/game/atoms.dm

```diff
@@ -1484,19 +1484,29 @@
 
 ///Sets the custom materials for an item.
 /atom/proc/set_custom_materials(list/materials, multiplier = 1)
```

Comment:
I don't like how much beauty code is suddenly in set_custom_materials, material behavior should be inside on removed and on apply. If this gets added it sets a precedent for doing things like this in set_custom_materials.